gordo/machine/model/anomaly/diff.py

```diff
@@ -159,14 +208,16 @@ def cross_validate(
         # Depend on having the trained fold models
         kwargs.update(dict(return_estimator=True, cv=cv))
 
-        cv_output = cross_validate(self, X=X, y=y, **kwargs)
+        cv_output = c_val(self, X=X, y=y, **kwargs)
 
         self.feature_thresholds_per_fold_ = pd.DataFrame()
         self.aggregate_thresholds_per_fold_ = {}
         self.smooth_feature_thresholds_per_fold_ = pd.DataFrame()
         self.smooth_aggregate_thresholds_per_fold_ = {}
         smooth_aggregate_threshold_fold = None
         smooth_tag_thresholds_fold = None
+        if self.window is not None and self.smoothing_method is None:
+            self.smoothing_method = "smm"
```

Comment:
Yes it could, my dilemma here is that defining a `smoothing_method` only makes sense if the `window` parameter exists, otherwise it should probably be set to `None`.
This was built like this to account for existing configs where smoothing method has not been defined since it wasn't available yet but `window` was defined.
If you strongly disagree I can indeed set it as default in the class definition.

Reply:
I see. No worries :) I guess there could be a default `window` parameter as well.

Reply:
Following @koropets suggestion I moved the whole `if` statement it to `__init__()`, which is the best compromise I guess, making sure `smoothing_method` is set if `window` exists when creating a new class instance, while not being a default. I guess this solves this?